extensions/blocks/podcast-player/view.js

```diff
@@ -0,0 +1,170 @@
+
+	if ( episodeLinkEl ) {
+		// Prevent handling clicks if a modifier is in use.
+		if ( e.shiftKey || e.metaKey || e.altKey ) {
+			return;
+		}
+
+		e.preventDefault();
+		handleEpisodeLinkClick( episodeLinkEl );
+	}
+} );
+
+function handleEpisodeLinkClick( episodeLinkEl ) {
+	// Get clicked episode element
+	const episodeEl = episodeLinkEl.closest( '.podcast-player__episode' );
+	if ( ! episodeEl ) {
+		// Append the error to closest parent if episode element is not present.
+		return renderEpisodeError( episodeLinkEl.closest( '*' ) );
+	}
+
+	// Get clicked episode audio URL
+	const audioUrl = episodeLinkEl.getAttribute( 'data-jetpack-podcast-audio' );
+	if ( ! audioUrl ) {
+		return renderEpisodeError( episodeEl );
+	}
+
+	// Get episode's parent block element
+	const blockEl = episodeEl.closest( '.wp-block-jetpack-podcast-player' );
+	if ( ! blockEl ) {
+		return renderEpisodeError( episodeEl );
+	}
+
+	// Get episode's state icon container element
+	const iconContainerEl = episodeEl.querySelector( '.podcast-player__episode-status-icon' );
+	if ( ! iconContainerEl ) {
+		return renderEpisodeError( episodeEl );
+	}
+
+	// Get player instance by block id
+	const player = playerInstances[ blockEl.id ];
+	if ( ! player ) {
+		return renderEpisodeError( episodeEl );
+	}
+
+	// Pause the player and set the state classes
+	player.audio.pause();
+	blockEl.classList.remove( 'is-playing' );
+	blockEl.classList.add( 'is-paused' );
+	iconContainerEl.innerHTML = ''; // remove the icon
+
+	// Get currently active episode element
+	const activeEpisodeEl = blockEl.querySelector( '.is-active' );
+
+	if ( activeEpisodeEl ) {
+		activeEpisodeEl.querySelector( '.podcast-player__episode-status-icon' ).innerHTML = '';
+		activeEpisodeEl.classList.remove( 'is-active' );
+		activeEpisodeEl
+			.querySelector( '[aria-pressed="true"]' )
+			.setAttribute( 'aria-pressed', 'false' );
+	}
+
+	player.audio.src = audioUrl;
+
+	player.audio
+		.play()
+		.then( function() {
+			blockEl.classList.remove( 'is-paused', 'is-error' );
+			blockEl.classList.add( 'is-playing' );
+			iconContainerEl.innerHTML =
+				'<svg xmlns="http://www.w3.org/2000/svg" height="24" viewBox="0 0 24 24" width="24"><path d="M0 0h24v24H0V0z" fill="none"/><path d="M3 9v6h4l5 5V4L7 9H3zm7-.17v6.34L7.83 13H5v-2h2.83L10 8.83zM16.5 12c0-1.77-1.02-3.29-2.5-4.03v8.05c1.48-.73 2.5-2.25 2.5-4.02zM14 3.23v2.06c2.89.86 5 3.54 5 6.71s-2.11 5.85-5 6.71v2.06c4.01-.91 7-4.49 7-8.77 0-4.28-2.99-7.86-7-8.77z"/></svg>';
+		} )
+		.catch( function() {
+			blockEl.classList.remove( 'is-playing', 'is-paused' );
+			blockEl.classList.add( 'is-error' );
+			iconContainerEl.innerHTML =
+				'<svg xmlns="http://www.w3.org/2000/svg" height="24" viewBox="0 0 24 24" width="24"><path d="M0 0h24v24H0V0z" fill="none"/><path d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"/></svg>';
```

Comment:
Rather than use these `<svg>`s inline, we could us these in the same way as the error where it's defined once in the DOM with `<symbol>` then utilize `<svg><use>` to render the correct icon.

Reply:
Oh I like this but we might have a sad christmas because of IE11 :( https://caniuse.com/#feat=mdn-api_svguseelement

Reply:
I think it'll be OK, as it seems like like the part of `<use>` we'd be... using... is OK: https://developer.mozilla.org/de/docs/Web/SVG/Element/use#Browser_compatibility

Reply:
Found that they're using in Jetpack already: https://github.com/Automattic/jetpack/blob/master/modules/widgets/social-icons.php#L404